python/cudf/cudf/core/frame.py

```diff
@@ -354,10 +354,13 @@ def interleave_columns(self):
         -------
         The interleaved columns as a single column
         """
+        if ('category' == self.dtypes)[0]:
```

Comment:
You can't just check the 0th result here, you need to check if any of them are `category`:
```suggestion
        if ('category' == self.dtypes).any():
```